File: src/Aurelia.js
import { class_type } from "./fable_modules/fable-library.4.0.0-theta-018/Reflection.js";
import { valueConverter, customElement } from "aurelia";

export class Decorators {
    constructor() {
    }
}

export function Decorators$reflection() {
    return class_type("Aurelia.Decorators", void 0, Decorators);
}

export function Decorators_CustomElement_4E60E31B(config) {
    return customElement(config);
}

export function Decorators_ValueConverter_Z721C83C5(name) {
    return valueConverter(name);
}

export class DecoratorsModule_JsonValueConverter {
    constructor() {
    }
    toView(value) {
        return JSON.stringify(value);
    }
}

export function DecoratorsModule_JsonValueConverter$reflection() {
    return class_type("Aurelia.DecoratorsModule.JsonValueConverter", void 0, DecoratorsModule_JsonValueConverter);
}

export function DecoratorsModule_JsonValueConverter_$ctor() {
    return new DecoratorsModule_JsonValueConverter();
}

Decorators_ValueConverter_Z721C83C5("json")(DecoratorsModule_JsonValueConverter);

File: src/Aurelia.fs
module Aurelia

open Fable.Core
open Fable.Core.JsInterop

[<Import("customElement", from = "aurelia")>]
let private customelement (config: obj) : 'T -> unit = jsNative

[<Import("valueConverter", from = "aurelia")>]
let private valueconverter (name: string) : 'T -> unit = jsNative

[<ImportDefault "aurelia">]
type Aurelia =

    static member register(obj) : Aurelia = jsNative
    member _.register(obj) : Aurelia = jsNative

    static member app(obj) : Aurelia = jsNative
    member _.app(obj) : Aurelia = jsNative

    static member start() : unit = jsNative
    member _.start() : unit = jsNative



type Decorators =

    static member CustomElement(config: obj) : obj -> unit = customelement config

    static member ValueConverter(name: string) : obj -> unit = valueconverter name


module Decorators =
    [<AttachMembers>]
    type JsonValueConverter() =
        member _.toView(value: obj) = JS.JSON.stringify (value)

    Decorators.ValueConverter ("json") jsConstructor<JsonValueConverter>
